Kubernetes-CLI.adoc

```diff
@@ -0,0 +1,63 @@
+= Kubernetes CLI
+
+The Infinispan CLI currently has two modes: command-line and interactive. 
+To improve the user experience when using the Infinispan Operator we can enhance the CLI to add Kubernetes capabilities.
+
+== Native
+
+Using Quarkus we can build the CLI as a native executable for the three most common platforms: Linux x86_64, OS X, Windows.
+
+== Kubernetes mode
+
+Installing the executable as `kubectl-infinispan` allows invocation as a `kubectl/oc` plugin, e.g.:
+
+`kubectl infinispan ...`
+
+Using the Fabric8 Kubernetes Client we can automatically detect the existence of a Kubernetes configuration (default to `~/.kube/config`) and automatically connect to the Kubernetes cluster.
+
+=== Kubernetes commands
+
+The Infinispan Kubernetes CLI plugin defaults to using the `datagrid` namespace and the `infinispan` service name. These can be overridden in each command using the `--namespace` parameter and the name argument.
+
+==== Install the operator
+
+`kubectl infinispan install [-n namespace]`
+
+Install the Infinispan Operator in the specified namespace. 
+This should also install the relevant OLM subscription and operator group so that the operator can be updated.
+
+==== Create an Infinispan service
+
+`kubectl infinispan create service [-n namespace] [-r replicas] [--expose-type=type] [--expose-port=port] [--expose-host=hostname] [servicename]`
```

Comment:
What type of service will this command create, cache or datagrid? The default by operator is `cache` but that has pretty much limited capabilities. Maybe having `[-t serviceType]` option here wouldn't be bad.

Also I'm trying to think whether `service` is a good verb in here. Even though I get why it was picked up, it pretty much fights with standard OpenShift `service` concept (and we might want it for potential service customization in the future?). Have you considered `cluster` or `server`? As `Infinispan Cluster` is used for referencing infinispan resources in the console there would be some consistency.

Other options could be `kubectl infinispan create infinispan` or not using a verb at all. Former doesn't feel exactly catchy and the later will probably cause issues maintaining different CRDs.

Reply:
Since it doesn't allow auth specification it'd be awesome if the create command would output the login details.

Reply:
I'll use cluster. Caches will be created using `kubectl infinispan create cache`